### ISSUE TITLE
builtin dark themes don't report themselves as dark (?)

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#3532]
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.2.997
    OS Version      : macOS 10.14
    R Version       : 3.5.1

### Description

Compare the output of `library(tidyverse)` between RStudio v1.1 and RStudio v1.2, with the builtin 'Tomorrow Night Bright' theme selected:

#### RStudio v1.1

<img width="587" alt="screen shot 2018-09-24 at 11 47 14 am" src="https://user-images.githubusercontent.com/1976582/45972080-9b655b00-bfef-11e8-86a3-fe9b5a12a476.png">

#### RStudio v1.2

<img width="584" alt="screen shot 2018-09-24 at 11 46 36 am" src="https://user-images.githubusercontent.com/1976582/45972084-9e604b80-bfef-11e8-9804-d868e2ab3130.png">

Note that the text is no longer white in RStudio v1.2.

(originally filed at https://github.com/tidyverse/tidyverse/issues/156)

**Comment by kevinushey** [comment]
It looks like `rstudioapi` does not report the correct theme:

```R
> rstudioapi::getThemeInfo()
$editor
[1] "TextMate"

$global
[1] "Modern"

$dark
[1] FALSE

$background
[1] "#FFFFFF"

$color
[1] "black"
```

This is even though I have Tomorrow Night Bright set as the theme.

**Comment by kevinushey** [comment]
It looks like RStudio is querying the 'theme' parameter, even though the theme is written to 'rstheme':

```
> .Call("rs_readUiPref", "theme")
NULL
> .Call("rs_readUiPref", "flat_theme")
[1] "default"
> .Call("rs_readUiPref", "rstheme")
$isDark
[1] TRUE

$name
[1] "Tomorrow Night Bright"

$url
[1] "theme/default/tomorrow_night_bright.rstheme"
```

See: https://github.com/rstudio/rstudio/blob/master/src/cpp/r/R/Api.R#L709

**Comment by batpigandme** [comment]
blurg! This is so weird, now the custom theme doesn't report back as dark…

```
> rstudioapi::getThemeInfo()
$editor
[1] "Night Owl"

$global
[1] "Modern"

$dark
[1] FALSE
```
Worked in the daily 9 days ago…
https://github.com/tidyverse/tidyverse/issues/156
![](https://user-images.githubusercontent.com/831732/45969880-0e2c0180-c003-11e8-9466-d7f194d264aa.png)

It's only messing with the tidyverse output, though, so I'll see if I can get it running right!

**Comment by batpigandme** [comment]
Ooh, ready for bonus weirdness? I didn't change the theme in between calls, but it reports as Tomorrow Night Bright for the first.

```r
> .Call("rs_readUiPref", "theme")
[1] "Tomorrow Night Bright"
> .Call("rs_readUiPref", "flat_theme")
[1] "default"
> .Call("rs_readUiPref", "rstheme")
$isDark
[1] TRUE

$name
[1] "Night Owl"

$url
[1] "theme/custom/local/night-owlish.rstheme"
```

**Comment by kevinushey** [comment]
I _think_ the 'theme' preference is supposed to be defunct (we only use 'rstheme' now), so it could indeed be stale and represent an older value. The fact that this custom theme is not reporting itself as dark is surprising, though!

Can we confirm whether the issue is truly that `rstudioapi::getThemeInfo()` and `.Call("rs_readUiPref", "rstheme")` are out-of-sync?

**Comment by batpigandme** [comment]
So they're only out of sync on `$isDark`. It affects very few elements, though. Like, if I search in the Help pane "No Results Found" is in white — actually, the only thing I can find that it really impacts is the tidyverse loading message (other conflicts are `.ace_constant .ace_language`, which is the style that's overridden w/ the tidyverse loading message).
```r
> rstudioapi::getThemeInfo()
$editor
[1] "Night Owl"

$global
[1] "Modern"

$dark
[1] FALSE

> .Call("rs_readUiPref", "rstheme")
$isDark
[1] TRUE

$name
[1] "Night Owl"

$url
[1] "theme/custom/local/night-owlish.rstheme"
```

**Comment by kevinushey** [comment]
@MariaSemple it looks like we hard-code a lot of these values according to the built-in theme being used:

https://github.com/rstudio/rstudio/blob/master/src/cpp/r/R/Api.R#L725-L784

Should we instead be reading these from the chosen theme directly?

**Comment by batpigandme** [comment]
Wooooo!! You're the best!
![image](https://user-images.githubusercontent.com/831732/46627498-a0a2c980-cb08-11e8-97db-2bb658f1a95d.png)

**Comment by batpigandme** [comment]
This is back in the daily builds. Oddly, the entire theme is being reported incorrectly. (I'm using Night Owl again, but it's coming back as `Textmate (default)`). The theme for RStudio itself is also being reported incorrectly (I have modern set, but it's reporting back as `"Classic"`
```r
rstudioapi::getThemeInfo()
$editor
[1] "Textmate (default)"

$global
[1] "Classic"

$dark
[1] FALSE

$foreground
[1] "rgb(214, 222, 235)"

$background
[1] "rgb(1, 22, 39)"
```

![image](https://user-images.githubusercontent.com/831732/64961905-77d09b80-d864-11e9-993d-4a3155806c28.png)

![image](https://user-images.githubusercontent.com/831732/64961844-55d71900-d864-11e9-9066-1eb851343d8e.png)

Let me know if you want me to open a fresh issue for this, since it seems slightly different to this one from last time.

**Comment by jmcphers** [comment]
I'd be willing to bet this is because of the new theme setting storage.

**Comment by dfalty** [comment]
Verified fixed 1.3.867
